lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/GeneratingSuggester.java

```diff
@@ -51,22 +51,22 @@
   }
 
   List<String> suggest(String word, WordCase originalCase, Set<String> prevSuggestions) {
-    List<Weighted<DictEntry>> roots = findSimilarDictionaryEntries(word, originalCase);
+    List<Weighted<Root<String>>> roots = findSimilarDictionaryEntries(word, originalCase);
```

Comment:
DictEntry is now parameterized, renamed to Root and moved into a separate file

Reply:
Thanks, these comment help a lot.